### PR TITLE
feat: record events on resources un/instrumented by the Lumigo operator

### LINKED ISSUE #24: Record events on injected resources
As a user, I want to have means of understanding why a resource injected by the Lumigo Operator is different than the resource I created with my IaC tooling.

When the controller injects into or removes injection from a resource, it should [record an event](https://book-v1.book.kubebuilder.io/beyond_basics/creating_events.html) on that resource.

### PR DESCRIPTION
**Pull request by mmanciop**
Closes #24

### CODE REVIEW

**Comment by mmanciop** [comment]
On second thought, we should also add the events over changes applied by the webhook. Doing it on `UPDATE` should not be hard, I have no idea yet how to do it for `CREATE` events :D

File: tests/e2e_suite_test.go [hunk 1]
```diff
@@ -682,14 +721,81 @@ func (m *matchesDeploymentCondition) Match(actual interface{}) (success bool, er
...
+}
+
+// eventList, err := clientset.CoreV1().Events(namespaceName).List(ctx, metav1.ListOptions{})
+// g.Expect(err).NotTo(HaveOccurred())
+
+// var addedLumigoInstrumentationEvent corev1.Event
+
+// g.Expect(addedLumigoInstrumentationEvent).NotTo(BeNil())
+// g.Expect(addedLumigoInstrumentationEvent.Source.Component).To(HavePrefix("lumigo-operator.v"))
+
```
Comment (GuyMoses) [inline tests/e2e_suite_test.go:798]: remove?

File: mutation/mutate.go [hunk 2]
```diff
@@ -204,15 +212,21 @@ func addAutoTraceLabel(objectMeta *metav1.ObjectMeta, value string) {
...
+func (m *mutatorImpl) removeLumigoFrom(topLevelObjectMeta *metav1.ObjectMeta, podTemplateSpec *corev1.PodTemplateSpec) (bool, error) {
+	originalSpec := podTemplateSpec.Spec.DeepCopy()
+
 	if err := m.removeLumigoFromPodSpec(&podTemplateSpec.Spec); err != nil {
-		return err
+		return false, err
+	}
+
+	if reflect.DeepEqual(originalSpec, &podTemplateSpec.Spec) {
+		return false, nil
```
Comment (GuyMoses) [inline mutation/mutate.go:223]: maybe give some kind of an error here? "remove didn't do anything?"

**Review by GuyMoses — APPROVED** [review]

File: mutation/mutate.go (hunk 2 above)
Comment (mmanciop) [reply mutation/mutate.go:223]: Yes, good point. Done in 58b56e5.

File: tests/e2e_suite_test.go (hunk 1 above)
Comment (mmanciop) [reply tests/e2e_suite_test.go]: Done in b42fe54

**Pull request merged**